server/src/lib/model/leaderboard.ts

```diff
@@ -205,9 +205,9 @@ async function getTopTeams(challenge: ChallengeToken): Promise<any[]> {
               @point1 := w1_points AS w1_points
             FROM (
                 SELECT teams.id, teams.name,
-                    SUM((earned_at BETWEEN challenges.start_date AND TIMESTAMPADD(WEEK, 2, challenges.start_date)) * points) AS w1_points,
-                    SUM((earned_at BETWEEN challenges.start_date AND TIMESTAMPADD(WEEK, 3, challenges.start_date)) * points) AS w2_points,
-                    SUM((earned_at BETWEEN challenges.start_date AND TIMESTAMPADD(WEEK, 4, challenges.start_date)) * points) AS w3_points
+                    SUM(((earned_at BETWEEN challenges.start_date AND TIMESTAMPADD(WEEK, 1, challenges.start_date) AND achievements.week IS NULL) OR achievements.week = 1) * points) AS w1_points,
+                    SUM(((earned_at BETWEEN challenges.start_date AND TIMESTAMPADD(WEEK, 2, challenges.start_date) AND achievements.week IS NULL) OR achievements.week = 2) * points) AS w2_points,
+                    SUM(((earned_at BETWEEN challenges.start_date AND TIMESTAMPADD(WEEK, 3, challenges.start_date) AND achievements.week IS NULL) OR achievements.week = 3) * points) AS w3_points
```

Comment:
First look at this SQL, it seems perfect.
But, I just got the chance to run the SQL locally, and found that some team may get 0 points in week 2, even though it had some achievements in week 1.
The `AND achievements.week IS NULL` shall ignore those achievements that are not meant for current week.
For example, in week 2, the week-1-only achievements will be ignored; in week 3, the week-1-only and week-2-only achievements will be ignored.

I thought in week 2, team points should include those that are achieved in week 1. Is it still the requirement?